Sources/StreamChat/Controllers/EntityDatabaseObserver.swift

```diff
@@ -284,3 +363,87 @@ private extension ListChangeAggregator {
         return self
     }
 }
+
+/// Observes changes of a single entity specified using an `NSFetchRequest`in the provided `NSManagedObjectContext`.
+class BackgroundEntityDatabaseObserver<Item, DTO: NSManagedObject>: BackgroundDatabaseObserver<Item, DTO> {
+    var item: Item? {
+        rawItems.first
+    }
+
+    /// Called with the aggregated changes after the internal `NSFetchResultsController` calls `controllerDidChangeContent`
+    /// on its delegate.
+    private var listeners: [(EntityChange<Item>) -> Void] = []
+
+    /// Creates a new `ListObserver`.
+    ///
+    /// Please note that no updates are reported until you call `startUpdating`.
+    ///
+    /// - Important: ⚠️ Because the observer uses `NSFetchedResultsController` to observe the entity in the DB, it's required
```

Comment:
What happens if there's no sort descriptor?

Reply:
A nice crash 😄 

```
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'An instance of NSFetchedResultsController requires a fetch request with sort descriptors'
```

DatabaseObservers are internal classes, and therefore it is us controlling its usage.